helm/external-dns-app/templates/deployment.yaml

```diff
@@ -19,84 +19,99 @@ spec:
         command:
         - /bin/sh
         - -c
-        - counter=5; while ! wget -qO- http://169.254.169.254/latest/meta-data/iam/security-credentials/ | grep {{ .Values.clusterID}}-Route53Manager-Role; do echo 'Waiting for iam-role to be available...'; sleep 5; let "counter-=1"  ; if [ "$counter" -eq "0" ]; then exit 1; fi; done
+        - counter=5; while ! wget -qO- http://169.254.169.254/latest/meta-data/iam/security-credentials/ | grep {{ template "aws.iam.role" . }}; do echo 'Waiting for iam-role to be available...'; sleep 5; let "counter-=1"  ; if [ "$counter" -eq "0" ]; then exit 1; fi; done
       {{- end }}
       containers:
-      - name: {{ .Values.name }}
-        image: "{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+      - name: {{ .Release.Name }}
+        image: "{{ .Values.global.image.registry }}/{{ .Values.global.image.name }}:{{ .Values.global.image.tag }}"
         imagePullPolicy: IfNotPresent
-        env:
         {{- if eq .Values.aws.access "external" }}
+        env:
         - name: AWS_ACCESS_KEY_ID
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.name }}-route53-credentials
+              name: {{ .Release.Name }}-route53-credentials
               key: aws_access_key_id
         - name: AWS_SECRET_ACCESS_KEY
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.name }}-route53-credentials
+              name: {{ .Release.Name }}-route53-credentials
               key: aws_secret_access_key
         - name: AWS_DEFAULT_REGION
           value: {{ .Values.aws.region }}
         {{- end }}
         args:
-        {{- range .Values.sources }}
+        {{- range .Values.externalDNS.sources }}
         - --source={{ . }}
         {{- end }}
+        {{- if .Values.externalDNS.annotationFilter }}
+        - --annotation-filter={{ .Values.externalDNS.annotationFilter }}
+        {{- end }}
+        {{- if .Values.externalDNS.interval }}
+        - --interval={{ .Values.externalDNS.interval }}
+        {{- end }}
         - --provider={{ .Values.provider }}
         {{- if eq .Values.provider "azure" }}
         - --azure-config-file=/azure/config/azure.yaml
-        {{- end }}
-        {{- if eq .Values.aws.access "external" }}
-        - --domain-filter={{ .Values.aws.baseDomain }}
-        - --aws-zone-type=public
+        {{- end -}}
+        {{- if eq .Values.provider "aws" }}
+        {{- include "zone.type" . | nindent 8 }}
+        {{- if .Values.aws.preferCNAME }}
         - --aws-prefer-cname
-        {{- else }}
-        - --domain-filter={{ .Values.baseDomain }}
         {{- end }}
-        - --metrics-address=:{{ .Values.metricsPort }}
-        - --namespace={{ .Values.namespace }}
+        {{- if .Values.aws.batchChangeSize }}
+        - --aws-batch-change-size={{ .Values.aws.batchChangeSize }}
+        {{- end }}
+        {{- include "domain.list" . | nindent 8 }}
```

Comment:
domain filter list is assembled in the helpers file so we just template the yaml in.